File: app/src/main/java/org/neidhardt/dynamicsoundboard/dialog/addnewsoundfromintent/CustomSpinner.java
package org.neidhardt.dynamicsoundboard.dialog.addnewsoundfromintent;

import android.content.Context;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.ArrayAdapter;
import android.widget.LinearLayout;
import android.widget.Spinner;
import org.neidhardt.dynamicsoundboard.R;

import java.util.List;


public class CustomSpinner extends LinearLayout implements View.OnClickListener
{
	private Spinner spinner;

	public CustomSpinner(Context context, AttributeSet attrs)
	{
		super(context, attrs);
		LayoutInflater.from(context).inflate(R.layout.view_spinner, this, true);
		this.spinner = (Spinner)this.findViewById(R.id.spinner);
		this.setOnClickListener(this);
	}

	public void setItems(List<String> items)
	{
		ArrayAdapter<String> adapter = new ArrayAdapter<>(getContext(), android.R.layout.simple_spinner_item, items);
		adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
		this.spinner.setAdapter(adapter);
	}

	public int getSelectedItemPosition()
	{
		return this.spinner.getSelectedItemPosition();
	}

	@Override
	public void onClick(View v)
	{
		this.spinner.performClick();
	}
}

File: ui_utils/src/main/kotlin/org/neidhardt/ui_utils/views/CustomSpinner.kt
package org.neidhardt.ui_utils.views

import android.R
import android.content.Context
import android.util.AttributeSet
import android.view.LayoutInflater
import android.view.View
import android.widget.ArrayAdapter
import android.widget.LinearLayout
import android.widget.Spinner
import org.neidhardt.dynamicsoundboard.R


class CustomSpinner(context: Context, attrs: AttributeSet) : LinearLayout(context, attrs), View.OnClickListener
{
	private val spinner: Spinner

	init
	{
		LayoutInflater.from(context).inflate(org.neidhardt.dynamicsoundboard.R.layout.view_spinner, this, true)
		this.spinner = this.findViewById(org.neidhardt.dynamicsoundboard.R.id.spinner) as Spinner
		this.setOnClickListener(this)
	}

	fun setItems(items: List<String>)
	{
		val adapter = ArrayAdapter(context, R.layout.simple_spinner_item, items)
		adapter.setDropDownViewResource(R.layout.simple_spinner_dropdown_item)
		this.spinner.adapter = adapter
	}

	val selectedItemPosition: Int
		get() = this.spinner.selectedItemPosition

	fun setSelectedItem(position: Int): Any = this.spinner.setSelection(position)

	override fun onClick(v: View)
	{
		this.spinner.performClick()
	}
}
